### PR TITLE
Feature/ticket 107/library borrows

### PR DESCRIPTION
**Pull request by RanaTayyab**

### CODE REVIEW

File: src/db/oracledb/books-dao.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ const getBookById = async (id) => {
   const query = `
     SELECT *
     FROM library_api_books
-    WHERE book_id = :bookId
+    WHERE bookid = :bookId
```
Comment (tsoliangwu0130) [inline src/db/oracledb/books-dao.js]: Hmm, why is the column name changed? In addition, I don't think `bookid` is following any naming convention, so why do we want to change it from `book_id` to `bookid`?

File: src/db/oracledb/borrows-dao.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+import { parseQuery } from 'utils/parse-query';
+import {
+  generateWhereClause,
+  generatePaginationParams,
+  convertKeysToLowercase,
+  formatDateInWhereClause,
+  convertDatesInObject,
+} from 'utils/dao-helper';
+
```
Comment (tsoliangwu0130) [inline src/db/oracledb/borrows-dao.js:15]: Unnecessary extra blank line here.

File: src/db/oracledb/borrows-dao.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+ * @param {string} id Unique borrow ID
+ * @returns {Promise<object>} Promise object represents a specific borrow or return undefined
+ *                            if term is not found
+ */
+const getBorrowById = async (id) => {
+  const connection = await getConnection();
+  const query = `
+    SELECT *
+    FROM library_api_borrows
+    WHERE borrowid = :borrowId
```
Comment (tsoliangwu0130) [inline src/db/oracledb/borrows-dao.js]: Same naming convention issue with `borrowid` here as well.

File: src/db/oracledb/borrows-dao.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+ * @param {object} body Request body
+ * @returns {Promise<object>} Promise object represents the posted borrow transaction
+ */
+const postBorrow = async (body) => {
+  const connection = await getConnection();
+  let response = {};
+
+  try {
+    const newBorrowDataRaw = body.data.attributes;
+    const currentPSTDate = new Date().toLocaleDateString('en-US', { timeZone: 'America/Los_Angeles' });
```
Comment (tsoliangwu0130) [inline src/db/oracledb/borrows-dao.js]: `currentPSTDate ` -> `currentPstDate`

**Review by tsoliangwu0130 — CHANGES_REQUESTED** [review]

**Comment by tsoliangwu0130** [comment]
Overall looks pretty good. I left a few styling comments.

File: src/db/oracledb/books-dao.js (hunk 1 above)
Comment (RanaTayyab) [reply src/db/oracledb/books-dao.js]: fixed.

File: src/db/oracledb/borrows-dao.js (hunk 2 above)
Comment (RanaTayyab) [reply src/db/oracledb/borrows-dao.js:15]: fixed.

File: src/db/oracledb/borrows-dao.js (hunk 3 above)
Comment (RanaTayyab) [reply src/db/oracledb/borrows-dao.js]: fixed.

File: src/db/oracledb/borrows-dao.js (hunk 4 above)
Comment (RanaTayyab) [reply src/db/oracledb/borrows-dao.js]: fixed.

File: src/utils/dao-helper.js [hunk 5]
```diff
@@ -133,11 +146,52 @@ const convertDatesInObject = (data) => {
...
+ * Converts all keys in an array of objects from snake_case or UPPERCASE to camelCase.
+ *
+ * @param {Array<Object>} rows - The array of objects with keys to be converted.
+ * @returns {Array<Object>} A new array of objects with keys converted to camelCase.
+ */
+const convertKeysToCamelCase = (rows) => rows.map((row) => Object.keys(row).reduce((acc, key) => {
+  const camelCaseKey = key.toLowerCase().replace(/_([a-z])/g, (match, letter) => letter.toUpperCase());
+  acc[camelCaseKey] = row[key];
+  return acc;
+}, {}));
```
Comment (tsoliangwu0130) [inline src/utils/dao-helper.js:161]: You can simply use this without reinventing the wheel: https://lodash.com/docs/4.17.15#camelCase

File: src/utils/dao-helper.js [hunk 6]
```diff
@@ -133,11 +146,52 @@ const convertDatesInObject = (data) => {
...
+ */
+const convertToSnakeCase = (obj) => Object.keys(obj).reduce((acc, key) => {
+  if (key === 'page[number]' || key === 'page[size]') {
+    acc[key] = obj[key];
+  } else {
+    const snakeCaseKey = key.replace(/([A-Z])/g, '_$1').toLowerCase();
+    acc[snakeCaseKey] = obj[key];
+  }
+  return acc;
+}, {});
```
Comment (tsoliangwu0130) [inline src/utils/dao-helper.js:178]: Simply use this: https://lodash.com/docs/4.17.15#snakeCase

**Review by tsoliangwu0130 — CHANGES_REQUESTED** [review]

**Comment by tsoliangwu0130** [comment]
Minor comments left.

File: src/utils/dao-helper.js (hunk 5 above)
Comment (RanaTayyab) [reply src/utils/dao-helper.js:161]: done

File: src/utils/dao-helper.js (hunk 6 above)
Comment (RanaTayyab) [reply src/utils/dao-helper.js:178]: done

**Review by tsoliangwu0130 — APPROVED** [review]

**Pull request merged**